win32ss/gdi/ntgdi/freetype.c

```diff
@@ -715,6 +711,83 @@ InitFontSupport(VOID)
     return TRUE;
 }
 
+static VOID
+IntFreeFontCacheList(PLIST_ENTRY pHead)
+{
+    PLIST_ENTRY pEntry, pNextEntry;
+    PFONT_CACHE_ENTRY pFontCache;
+
+    ASSERT_FREETYPE_LOCK_HELD();
+
+    for (pEntry = pHead->Flink; pEntry != pHead; pEntry = pNextEntry)
+    {
+        pNextEntry = pEntry->Flink;
+        pFontCache = CONTAINING_RECORD(pEntry, FONT_CACHE_ENTRY, ListEntry);
+        RemoveCachedEntry(pFontCache);
+    }
+
+    pHead->Flink = pHead;
```

Comment:
The way @katahiromz currently wrote the code, he would need the pNextEntry variable because after RemoveCachedEntry is called, the pEntry pointer is invalid, and doing `pEntry->Flink` afterwards would be a bug.